### PR TITLE
[BACKLOG-32465] Hadoop Clusters - Embed Site XML files with Cluster C…

### PR DESCRIPTION
**Pull request by tkafalas**
…onfiguration

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 49s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: common-fragment-V1/src/main/java/org/pentaho/hadoop/shim/common/format/orc/PentahoOrcInputFormat.java [hunk 1]
```diff
@@ -55,7 +57,11 @@ public PentahoOrcInputFormat( NamedCluster namedCluster ) {
       conf = inClassloader( () -> {
         Configuration confProxy = new ConfigurationProxy();
         confProxy.addResource( "hive-site.xml" );
-        ShimConfigsLoader.addConfigsAsResources( namedCluster.getName(), confProxy::addResource );
+        BiConsumer<InputStream, String> consumer = ( is, filename ) ->
+        {
+          confProxy.addResource( is, filename );
+        };
+        ShimConfigsLoader.addConfigsAsResources( namedCluster, consumer );
```
Comment (peterrinehart) [inline common-fragment-V1/src/main/java/org/pentaho/hadoop/shim/common/format/orc/PentahoOrcInputFormat.java]: Can we roll this functionality directly into the NamedCluster object and start to eliminate ShimConfigsLoader?

File: common-fragment-V1/src/main/java/org/pentaho/hbase/shim/common/CommonHBaseConnection.java [hunk 2]
```diff
@@ -121,42 +123,46 @@ public static Integer getMajorVersionNumber( String shimConfigurationId ) {
...
 
       m_config = new Configuration();
+      BiConsumer<InputStream, String > consumer = ( is, filename ) ->
+      {
+        m_config.addResource( is, filename );
+      };
       try {
         if ( namedClusterIsMapr ) {
-          ShimConfigsLoader.addConfigsAsResources( namedCluster, m_config::addResource,
+          ShimConfigsLoader.addConfigsAsResources( namedCluster, consumer,
```
Comment (peterrinehart) [inline common-fragment-V1/src/main/java/org/pentaho/hbase/shim/common/CommonHBaseConnection.java]: Can this method be made a member of the NamedCluster object so we can get rid of ShimConfigsLoader?

File: common-fragment-V1/src/main/java/org/pentaho/hadoop/shim/common/format/orc/PentahoOrcInputFormat.java (hunk 1 above)
Comment (tkafalas) [reply common-fragment-V1/src/main/java/org/pentaho/hadoop/shim/common/format/orc/PentahoOrcInputFormat.java]: If we put that in namedcluster than we add Vendor Library dependencies to the NamedCluster, which has versions that have to compile in the kettle.

**Comment by buildguy** [comment]
### :x: Build failed in 46s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 15s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl common-fragment-V1,shim-api
```
---
:ok_hand: All tests passed!
 
Tests run: **911**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1120/10/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**